feat: opt-in alternate link consistency

divine opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)

Please carefully read the contribution docs before creating a pull request
 👉 https://nuxt.com/docs/community/contribution
-->

### 🔗 Linked issue

Fix #3296 

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Use canonical queries options for alternate links as well.
### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

divine committed:
fix: use canonicalQueries for alternate links

divine commented:
We need this fix as soon as possible. It's harming SEO.

You're free to modify the code and push any changes.

Thanks!

BobbieGoede commented:
Will take a look at this soon and will add tests to prevent regressions in the future

divine commented:
> Will take a look at this soon and will add tests to prevent regressions in the future

It's pretty important for our project. Any update?

Thanks!

BobbieGoede committed:
fix: remove query parameters for alternate links

BobbieGoede committed:
feat: add experimental `alternateLinkCanonicalQueries` property

BobbieGoede committed:
fix: do not append params twice

BobbieGoede committed:
test: assert `alternateLinkCanonicalQueries` functionality

BobbieGoede committed:
docs: document `experimental.alternateLinkCanonicalQueries`

BobbieGoede commented:
I have fixed the implementation and since this would be a breaking change I have made it an opt-in feature with the property `experimental.alternateLinkCanonicalQueries`.

I have updated your reproduction from #3296 to demonstrate it using the preview release here https://stackblitz.com/edit/bobbiegoede-nuxt-i18n-starter-kge99mlf?file=nuxt.config.ts, is this what you had in mind?

We may make this default behavior in the next major, but hopefully I can get some more conclusive docs/sources to be sure this is the right approach.

divine commented:
Hello,

> I have fixed the implementation and since this would be a breaking change I have made it an opt-in feature with the property `experimental.alternateLinkCanonicalQueries`.
> 
> I have updated your reproduction from #3296 to demonstrate it using the preview release here https://stackblitz.com/edit/bobbiegoede-nuxt-i18n-starter-kge99mlf?file=nuxt.config.ts, is this what you had in mind?

Exactly, thank you for fixing this.

> We may make this default behavior in the next major, but hopefully I can get some more conclusive docs/sources to be sure this is the right approach.

I've explained this in this comment https://github.com/nuxt-modules/i18n/issues/3296#issuecomment-2599674173

This is a screenshot from a Google video:

![image](https://github.com/user-attachments/assets/f2da787d-2cff-4638-a4c8-cba80c5142c4)

Google suggests stripping out all GET parameters except those needed as a page parameters.

This is the only correct way to handle SEO. Otherwise it's only harming.

Thanks!
